incubator/hnc/pkg/kubectl/root.go

```diff
@@ -135,10 +137,35 @@ func (cl *realClient) updateHierarchy(hier *api.HierarchyConfiguration, reason s
 	}
 }
 
+func (cl *realClient) hasNamespace(nnm string) bool {
+	if _, err := k8sClient.CoreV1().Namespaces().Get(nnm, metav1.GetOptions{}); err != nil {
+		if errors.IsNotFound(err) {
+			return false
+		}
+		fmt.Printf("Error reading namespace %s: %s\n", nnm, err)
+		os.Exit(1)
+	}
+	return true
+}
+
 func (cl *realClient) createHierarchicalNamespace(nnm string, hnnm string) {
 	hns := &api.HierarchicalNamespace{}
 	hns.Name = hnnm
 	hns.Namespace = nnm
+	// Set initial hns state before creating the hns resource. "api.Forbidden" and "api.Conflict"
+	// are only set if the resource is created successfully bypassing the admission controller.
+	// TODO: ensure the specified child doesn't already exist and the parent is allowed
+	//  to create self-serve subnamespaces through the admission controller. See issue
+	//  https://github.com/kubernetes-sigs/multi-tenancy/issues/458
+	if reconcilers.EX[nnm] {
+		hns.Status.State = api.Forbidden
+	} else {
+		if cl.hasNamespace(hnnm) {
+			hns.Status.State = api.Conflict
+		} else {
+			hns.Status.State = api.Missing
+		}
+	}
```

Comment:
Clients are never supposed to set the status of an object, only the spec - only reconcilers can set the status, because they can change at any time. So I don't think you should make any changes to kubectl at all - the client is just doing what the user asked it to, and it's up to the admission controller to reject it, or the reconciler to mark it as invalid.